crates/autopilot/src/on_settlement_event.rs

```diff
@@ -306,3 +216,224 @@ impl Inner {
         }
     }
 }
+
+#[async_trait]
+impl OnSettlementEventAction for Updater {
+    async fn apply(&self) -> Result<bool> {
+        let mut ex = self
+            .db
+            .pool
+            .begin()
+            .await
+            .context("acquire DB connection")?;
+        let event = match database::settlements::get_settlement_without_auction(&mut ex)
```

Comment:
we also want this as a separate function under infra::Persistance but should probably be done in a separate PR.